assets/js/hocs/with-store-cart-api-hydration.js

```diff
@@ -13,13 +13,50 @@ import { useSelect } from '@wordpress/data';
  */
 const useStoreCartApiHydration = () => {
 	const cartData = useRef( getSetting( 'cartData' ) );
+	const { cartDataIsStale } = useDispatch( CART_STORE_KEY );
 
 	useSelect( ( select, registry ) => {
 		if ( ! cartData.current ) {
 			return;
 		}
+		const { isResolving, hasFinishedResolution, isCartDataStale } = select(
+			CART_STORE_KEY
+		);
 
-		const { isResolving, hasFinishedResolution } = select( CART_STORE_KEY );
+		/**
+		 * This should only execute once. When the code further down the file executes
+		 * then the condition directly below this comment should never evaluate to true
+		 * on subsequent executions. Because of this localStorage.getItem won't be
+		 * called multiple times.
+		 */
+		if (
+			! isCartDataStale() &&
+			! isResolving( 'getCartData' ) &&
+			! hasFinishedResolution( 'getCartData', [] )
+		) {
+			const lastCartUpdateRaw = window.localStorage.getItem(
+				'lastCartUpdate'
+			);
+			const lastCartUpdate =
+				lastCartUpdateRaw === null
+					? null
+					: JSON.parse( lastCartUpdateRaw );
+
+			if ( lastCartUpdate?.timestamp === undefined ) {
```

Comment:
If you have only one value you care about here you could directly store the timestamp and avoid working with objects. 
`lastCartUpdateTimestamp: X`

I also recommend to rename the key to `wc-blocks_cart_update_timestamp` to follow the naming pattern used for `wc-blocks_dismissed_compatibility_notices` that easily identifies the items coming from blocks.

Reply:
Thanks, I based it on the local storage key used to track the store API nonce update which is called `storeApiNonce` - do you think it's worth adding a cooldown issue to rename this to `wc-blocks_store_api_nonce`?

Reply:
sure, sounds good